plugins/modules/mount.py

```diff
@@ -639,6 +675,123 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     return mounts
 
 
+def is_swap(module, args):
+    """Return True if the device/file is an active swap space, False otherwise."""
+
+    if module.params['fstype'] != 'swap':
+        return False
+
+    if SYSTEM == 'sunos':
+        swap_bin = module.get_bin_path('swap', required=True)
+        cmd = [swap_bin, '-l']
+    elif SYSTEM.endswith('bsd'):
+        swapctl_bin = module.get_bin_path('swapctl', required=True)
+        cmd = [swapctl_bin, '-l']
+    else:
+        # swapon is supposed to be the standard command
+        swapon_bin = module.get_bin_path('swapon', required=True)
+        cmd = [swapon_bin]
+
+    rc, out, err = module.run_command(cmd)
+
+    if rc:
+        module.fail_json(msg="Error while querying active swaps: %s" % err)
+
+    # Get the first field of each line but the header
+    devices = [x.split()[0] for x in out.splitlines()[1:]]
+    dev = os.path.realpath(args['src'])
+
+    if SYSTEM == 'linux':
+        if args['src'].startswith('UUID='):
+            uuid_path = os.path.join('/dev/disk/by-uuid', args['src'].split('=')[1])
+            dev = os.path.realpath(uuid_path)
+        elif args['src'].startswith('LABEL='):
+            label_path = os.path.join('/dev/disk/by-label', args['src'].split('=')[1])
+            dev = os.path.realpath(label_path)
+    elif SYSTEM == 'freebsd':
+        if args['src'].startswith('md'):
+            dev = os.path.join('/dev', args['src'])
+
+    return bool(dev in devices)
+
+
+def swapon(module, args):
+    """Activate a swap device/file with the proper options."""
+
+    if SYSTEM == 'sunos':
+        swap_bin = module.get_bin_path('swap', required=True)
+        cmd = [swap_bin, '-a']
+    elif SYSTEM.endswith('bsd'):
+        swapctl_bin = module.get_bin_path('swapctl', required=True)
+        cmd = [swapctl_bin, '-a']
+    else:
+        # swapon is supposed to be the standard command, isn't it ?
+        swapon_bin = module.get_bin_path('swapon', required=True)
+        cmd = [swapon_bin]
```

Comment:
I would query swap binary path before and attach it with the `SYSTEM` variable so that it is not required to query each time.